app/code/Magento/Sales/Model/Order/Email/Sender/OrderSender.php

```diff
@@ -130,14 +131,17 @@ protected function prepareTemplate(Order $order)
             'formattedShippingAddress' => $this->getFormattedShippingAddress($order),
             'formattedBillingAddress' => $this->getFormattedBillingAddress($order),
         ];
-        $transport = new \Magento\Framework\DataObject($transport);
+        $transportObject = new DataObject($transport);
 
+        /**
+         * Event argument `transport` is @deprecated. Use `transportObject` instead.
+         */
         $this->eventManager->dispatch(
             'email_order_set_template_vars_before',
-            ['sender' => $this, 'transport' => $transport]
+            ['sender' => $this, 'transport' => $transportObject->getData(), 'transportObject' => $transportObject]
```

Comment:
Hi, @gwharton, due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't rename or remove event arguments or change their type.

Reply:
If you look at the original commit in #10210, the event arguments were changed for all email_***_set_template_vars_before events already. This PR, and the other two, are correcting the fact that the original author missed one of the 7 events that needed updating. 

If this PR and the other 2 don't go through, it will leave us with 6 of the events with different types to this one.

We add a new object 'transportObject' to the event, but maintain the original 'transport' object for backwards compatability.

Reply:
@gwharton As I see, in 10210, that fixes were delivered to Magento 2.1 and 2.2 branches but weren't delivered to 2.3
> We add a new object 'transportObject' to the event, but maintain the original 'transport' object for backwards compatability.
- But as I see the `transfer` object interface was changed or I was mistaken?

Reply:
Yes, the code to generate $transport has changed, but the contents and type of $transport have not changed.